reporter/datadog_reporter.go

```diff
@@ -350,6 +354,145 @@ func (r *DatadogReporter) reportProfile(ctx context.Context) error {
+						buildID = executionInfo.buildID
+					}
+				}
+
+				tmpMapping := createPprofMapping(profile, uint64(traceInfo.linenos[i]),
+					fileName, buildID)
+				fileIDtoMapping[traceInfo.files[i]] = tmpMapping
+				loc.Mapping = tmpMapping
+			}
+			line := pprofile.Line{Function: createPprofFunctionEntry(funcMap, profile, "",
+				loc.Mapping.File)}
+			loc.Line = append(loc.Line, line)
+		case libpf.KernelFrame:
+			// Reconstruct frameID
+			frameID := libpf.NewFrameID(traceInfo.files[i], traceInfo.linenos[i])
+			// Store Kernel frame information as Line message:
+			line := pprofile.Line{}
+
+			if tmpFunction, exists := frameIDtoFunction[frameID]; exists {
+				line.Function = tmpFunction
+			} else {
+				symbol, exists := r.fallbackSymbols.Get(frameID)
+				if !exists {
+					// TODO: choose a proper default value if the kernel symbol was not
+					// reported yet.
+					symbol = unknownStr
+				}
+				line.Function = createPprofFunctionEntry(
+					funcMap, profile, symbol, "")
+			}
+			loc.Line = append(loc.Line, line)
+
+			// To be compliant with the protocol generate a dummy mapping entry.
+			loc.Mapping = getDummyMapping(fileIDtoMapping, profile,
+				traceInfo.files[i])
+
+		case libpf.AbortFrame:
+			// Next step: Figure out how the OTLP protocol
+			// could handle artificial frames, like AbortFrame,
+			// that are not originate from a native or interpreted
+			// program.
+		default:
+			// Store interpreted frame information as Line message:
+			line := pprofile.Line{}
+
+			fileIDInfoLock, exists := r.frames.Get(traceInfo.files[i])
+			if !exists {
+				// At this point, we do not have enough information for the frame.
+				// Therefore, we report a dummy entry and use the interpreter as filename.
+				line.Function = createPprofFunctionEntry(funcMap, profile,
+					"UNREPORTED", frameKind.String())
+			} else {
+				fileIDInfo := fileIDInfoLock.RLock()
+				si, exists := (*fileIDInfo)[traceInfo.linenos[i]]
+				if !exists {
+					// At this point, we do not have enough information for the frame.
+					// Therefore, we report a dummy entry and use the interpreter as filename.
+					// To differentiate this case with the case where no information about
+					// the file ID is available at all, we use a different name for reported
+					// function.
+					line.Function = createPprofFunctionEntry(funcMap, profile,
+						"UNRESOLVED", frameKind.String())
+				} else {
+					line.Line = int64(si.lineNumber)
+
+					line.Function = createPprofFunctionEntry(funcMap, profile,
+						si.functionName, si.filePath)
+				}
+				fileIDInfoLock.RUnlock(&fileIDInfo)
+			}
+			loc.Line = append(loc.Line, line)
+
+			// To be compliant with the protocol generate a dummy mapping entry.
+			loc.Mapping = getDummyMapping(fileIDtoMapping, profile, traceInfo.files[i])
+		}
+		sample.Location = append(sample.Location, loc)
+	}
+
+	execPath, _ := r.execPathes.Get(traceInfo.pid)
+	baseExec := path.Base(execPath)
```

Comment:
Why not compute `baseExec` after kernel frame check since it can change `execPath` ?